ui/src/pages/QuizPage.vue

```diff
@@ -0,0 +1,424 @@
+              <i class="bi bi-bookmark-heart" />
+            </button>
+          </div>
+
+          <div class="modal-body">
+            <div class="row">
+              <div class="col-4">
+                <div class="text-center">
+                  <img
+                    v-if="isCorrect"
+                    src="../assets/correct.png"
+                    class="answer-img"
+                    alt=""
+                  />
+                  <img
+                    v-else
+                    src="../assets/miss.png"
+                    class="answer-img"
+                    alt=""
+                  />
+                </div>
+              </div>
+
+              <div class="col-8">
+                <div v-if="isCorrect">
+                  <h3 class="answer-title">正解！</h3>
+                  <p>正解は{{ correctAnswer }}！</p>
+                  <p></p>
+                </div>
+
+                <div v-else>
+                  <h3 class="answer-title">残念...!</h3>
+                  <p>正解は{{ correctAnswer }}！</p>
+                  <p></p>
+                </div>
+              </div>
+            </div>
+          </div>
+
+          <div class="modal-footer">
+            <button 
+              v-show="currentQuizNumber < totalQuizNumber" 
+              type="button" class="btn btn-primary" 
+              data-dismiss="modal" 
+              @click="toNextQuiz()"
+            >
+              次の問題へ
+            </button>
+            <button 
+              v-if="currentQuizNumber == totalQuizNumber" 
+              type="button" class="btn btn-primary" 
+              data-toggle="modal" data-target="#answerEndedModal" 
+              @click="toNextQuiz()"
+            >
+              結果へ
+            </button>
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <!-- すべての問題を回答し終わったときに表示するモーダル -->
+
+    <div
+      class="modal fade"
+      id="answerEndedModal"
+      tabindex="-1"
+      role="dialog"
+      aria-labelledby="exampleModalCenterTitle"
+      aria-hidden="true"
+    >
+      <div class="modal-dialog modal-dialog-centered" role="document">
+        <div class="modal-content">
+          <div class="modal-header">おつかれさまでした！</div>
+
+          <div class="modal-body">
+            <div class="row">
+              <div class="col-4">
+                <div class="text-center">
+                  <img src="../assets/trophy.png" class="answer-img" />
+                </div>
+              </div>
+
+              <div class="col-8">
+                <p>正答数は{{ correctNumber.length }}/{{ totalQuizNumber }}</p>
+                <div v-for="(result, number) in results" :key="result">
+                  <div v-if="result">第{{ number+1 }}問.〇</div>
+                  <div v-else>第{{ number+1 }}問.×</div>
+                </div>
+              </div>
+            </div>
+          </div>
+
+          <div class="modal-footer">
+            <button class="btn btn-success" @click="sendResult()">トップへ戻る</button>
```

Comment:
> 未実装箇所
>・ページ遷移、およびAPIへのデータの受け渡し（正しい記述がわからなかったため）

例えばだけど、こんな感じでいけると思う！

```suggestion
           <button class="btn btn-success" @click="toTopPage">トップへ戻る</button>
```
```js
const router = useRouter();

const toTopPage = () => {
    sendResult();
    router.push("/");
};
```